software/spack_env/spock_spack_env.rst

```diff
@@ -0,0 +1,136 @@
+
+The provided Spack environment files are intended to assist OLCF users in setup their development environment at the 
+OLCF.  The base environment file includes the compilers and packages that are installed at the system level.
+
+Traditionally, the user environment is modified by module files, so a user would add use  ``module load cmake/3.18.2`` to 
+load CMake version 3.18.2 into their environment.  Using a Spack environment, a user can add an OLCF provided package
+and build against it using Spack without having to load the module file separately.
+
+The information presented here is a subset of what can be found at `the Spack documentation site <https://spack.readthedocs.io/>`_.
+
+.. note::
+  Definitions
+
+  - Spack environment - A set of Spack specs for the purpose of building, rebuilding and deploying in a coherent fashion.
+
+  - External packages - An externally-installed package used by Spack, rather than building its own package.
+
+  - Chained instance - A Spack instance that uses any packages installed 
+
+Getting Started
+------------------------------------------------
+
+Clone the OLCF User Environment repo and the Spack repo, start a new Spack instance, and create and activate a new Spack environment:
+
+.. code-block:: none
+
+  ## From a Spock login
+
+  > git clone https://code.ornl.gov/2ff/olcf-user-environments.git
+  > cd olcf-user-environments
+
+  > git clone https://github.com/spack/spack.git
+  > source spack/share/spack/setup-env.sh
+
+  > spack env create my_env linux-sles15-zen2/spock/spack.yaml
+  > spack env activate my_env
+
+Now a user can add and install their dependencies with Spack and proceed with developing their application.
+
+Chained Spock Instances
+------------------------------------------------
+
+The Spock Spack environment includes this configuration addition:
+
+.. code-block:: none
+
+  upstreams:
+    olcf:
+      install_tree: /sw/spock/spack-envs/base/opt
+
+This allows the user's newly created instance to use any package installed into the above directory as if it was
+installed by the user.
+
+.. note::
+
+  NOTE ABOUT UPSTREAM SPACK INSTALLATIONS
+
+  This other instance of Spack has no knowledge of the local Spack instance and may not have the same permissions
+  or ownership as the local Spack instance. This has the following consequences:
+
+  - Upstream Spack instances are not locked. Therefore it is up to users to make sure that the local instance is not using an upstream instance when it is being modified.
+  - Users should not uninstall packages from the upstream instance. Since the upstream instance doesn’t know about the local instance, it cannot prevent the uninstallation of packages which the local instance depends on.
+
+  See `here <https://spack.readthedocs.io/en/latest/chain.html#chaining-spack-installations>`_ for more information
+  regarding chained Spack instances.
+
+Add Dependencies to the environment
+------------------------------------------------
+
```

Comment:
Does the spock spack not need a section on "Adding OLCF Installed Dependencies as External Packages" like we have for the Summit spack instructions? I'm not sure hence why I'm asking you to make sure.

Reply:
No, this is taken care of by chaining the instances.  Basically it gives access to everything already installed.  On summit we have to include them manually.